### PR TITLE
[Experimental] Adds URL scheme support for directly opening themes in NetNewsWire.

### PR DESCRIPTION
**Pull request by stuartbreckenridge**
## Add Themes Directly to NetNewsWire
Theme developers: on iOS and macOS, themes can be opened directly in NetNewsWire using the below URL scheme:

`netnewswire://theme/add?url={url}`

When using this URL scheme the theme being shared must be zipped.

Parameters:
- `url`: (mandatory, URL-encoded): The theme's location.

### CODE REVIEW

File: Shared/ArticleStyles/ArticleThemePlist.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	public var version: Int
+	
+	enum CodingKeys: String, CodingKey {
+		case name = "Name"
+		case themeIdentifier = "ThemeIdentifer" // FIXME: Spelling error!
+		case creatorHomePage = "CreatorHomePage"
+		case creatorName = "CreatorName"
+		case version = "Version"
+	}
+}
```
Comment (vincode-io) [inline Shared/ArticleStyles/ArticleThemePlist.swift:25]: If we are going to add this, we should go ahead and change `ArticleTheme` to use it to remove redundant definitions. Also, let's just go ahead and fix the typo.

Comment (stuartbreckenridge) [reply Shared/ArticleStyles/ArticleThemePlist.swift:25]: This has been implemented for `ArticleTheme`.

File: iOS/SceneDelegate.swift [hunk 2]
```diff
@@ -165,11 +168,56 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
+	
+	
+	private func showAlert(_ error: Error) {
+		let alert = UIAlertController(title: NSLocalizedString("Error", comment: "Error"),
+									  message: error.localizedDescription,
+									  preferredStyle: .alert)
+		alert.addAction(UIAlertAction(title: NSLocalizedString("Dismiss", comment: "Dismiss"), style: .cancel, handler: nil))
+		self.window?.rootViewController?.present(alert, animated: true, completion: nil)
+	}
 }
```
[the remark this answers is not in the document]
Comment (stuartbreckenridge) [reply iOS/SceneDelegate.swift:207]: This is now being used.

**Comment by stuartbreckenridge** [comment]
@vincode-io the last few commits should largely resolve the above feedback.

**Pull request merged**